### DISCUSSION TITLE
Field sampling using FieldSet.from_netcdf

### DISCUSSION THREAD

**Discussion by MiriamSterl** [Parcels-code/Parcels#1257] [Q&A] [answered]
I want to sample the velocity fields of the particles in my simulation, as well as an additional field, the potential vorticity. I have the PV field for each time saved in the same netCDF file that contains my velocity fields. How can I add this PV field to my fieldset? I have looked at the tutorial on field sampling (https://nbviewer.org/github/OceanParcels/parcels/blob/master/parcels/examples/tutorial_sampling.ipynb?flush_cache=true), but it seems that the `extra_fields` keyword they used there can only be used when creating a fieldset using `FieldSet.from_parcels`. However, I use `FieldSet.from_netcdf`, which doesn't have this keyword. How can I sample additional fields in this case?

**Comment by erikvansebille** [comment, marked as answer]
You can simply add fields in the dictionary of variables. For example, if you have a field called PV

```python
filenames = {'U': fnameU, 'V': fnameV, 'PV': fnamePV}
variables = {'U': 'u', 'V': 'v', 'PV': 'PV'}
dimensions = {'lat': 'lat', 'lon': 'lon'}
set = FieldSet.from_netcdf(filenames, variables, dimensions)
```

The resulting `fset` now has three fields: `U`, `V` and `PV`. You can do this for how many fields you want, and by making dimensions a dict-of-dicts (i.e. a separate dictionary for each variable), you can even combine fields with different dimensions.

**Reply by MiriamSterl** [reply]
Thanks!
